File: IntelligentRecruitment_microservice/application-service/src/main/java/com/fedicode/applicationservice/Dto/AdminApplicationResponse.java
package com.fedicode.applicationservice.Dto;

import com.fedicode.applicationservice.Entity.Application;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Builder
@Data
public class AdminApplicationResponse {
    int id;
    String firstName;
    String lastName;
    String email;
    String phone;
    BigDecimal llmScore;
    String llmAnalysis;
    String cvFilePath;
    LocalDateTime createdAt;
    Application.ApplicationStatus status;
    //Job-details
    String title;
    String description;
    String location;
    LocalDateTime jobCreatedAt;

    //Recruiter-details
    String recruiterFirstName;
    String recruiterLastName;
    String recruiterEmail;

}

File: IntelligentRecruitment-UI/src/app/models/Dto/Application/AdminApplicationResponse.ts
import {ApplicationStatus} from '../../Enum/ApplicationStatus';

export interface AdminApplicationResponse {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  llmScore: number | null;
  llmAnalysis: string | null;
  cvFilePath: string | null;
  createdAt: string;
  status: ApplicationStatus;
  jobId: number;
  //Job-details
  title: string | null;
  description?: string | null;
  location: string | null;
  jobCreatedAt: string | null;
  //Recruiter-details
  recruiterFirstName: string | null;
  recruiterLastName: string | null;
  recruiterEmail: string | null;
}
